### ISSUE TITLE
Ember auth doesn't work in non-restful resources 

### ISSUE THREAD

**Issue by seifsallam** [heartsentwined/ember-auth#78]
I have a method where i need to vote for a post using `user_id`, but when sending POST request to backend, user `auth_token` doesn't get appended, knowing that it works in all other requests.

``` javascript
App.Post.reopenClass({
  vote: function(post_id) {
    $.ajax({
      url: "posts/" + post_id + "/votes",
      type: "POST",
      success: function(data, textStatus, xhr) {
        console.log("Voted");
      },
      error: function(xhr, textStatus, errorThrown) {
        return alert("error");
      }
    });
  }
});
```

Is there anything that i need to do? or do i have to manually append all requests with `auth_token`

**Comment by heartsentwined** [comment]
If you are using the `jquery` `requestAdapter` (which is the default), then `ember-auth` has this function in-built.

Relevant section from the [docs](http://ember-auth.herokuapp.com/):

> I want to send an authenticated request to my server, like calling [jQuery.ajax(settings)](http://api.jquery.com/jQuery.ajax/), but with authentication information from the current user appended (if any).
> 
> ```
> App.Auth.send(settings)
> ```

So change your code slightly with this:

``` javascript
App.Post.reopenClass({
  vote: function(post_id) {
    App.Auth.send({
      url: "posts/" + post_id + "/votes",
      type: "POST",
      success: function(data, textStatus, xhr) {
        console.log("Voted");
      },
      error: function(xhr, textStatus, errorThrown) {
        return alert("error");
      }
    });
  }
});
```

However, if you are not using `jquery` `requestAdapter`, then you'll have to include the auth token on your own.

**Comment by seifsallam** [comment]
Thanks, working great.

**Comment by flynfish** [comment]
I am having a similar issue, but its when I am just calling model.save(). Am I doing something wrong? This worked fine in 7.1.0 but I just updated to 7.1.2 and it stopped appending the auth_token to the PUT request. 

```
var model = this.get("model");
        model.set("plan", plan);
        model.save().then(this.updatedPlan.bind(this), this.updatedPlanDidError.bind(this));
```

**Comment by flynfish** [comment]
I am also using EmberModel, v 0.0.8 so not sure if there is an issue with that. I will try to debug but if you have any ideas on tracking down the issue please let me know!

**Comment by heartsentwined** [comment]
@flynfish I'm totally puzzled with the observed behavior. Here is the complete changeset between `7.1.0` and `7.1.2`:

Relevant / major:
- `emberModel` rewritten, to support a changed API from `ember-model`'s end - see #84. The change is introduced in `ember-model` `v0.0.7`. Affects `emberModel` module users.
- `$.url` / `purl` library updated from `v2.2.1` to `v2.3.1`. Affects `urlAuthenticatable` module users.

Remaining stuff / misc:
- typo fix in `dummy` `requestAdapter` (#86)
- added ruby-version ignores
- more ember version tests
- a couple of gem updates - for tests

I'd guess it's actually something related to ember-model changes. Or maybe the rewritten `emberModel` doesn't actually cover all use cases - in that case, sorry, I have zero experience in the `emberModel` code base, we'd need PRs from some `ember-model` user.

**Comment by heartsentwined** [comment]
Reopening, although this appears to be a separate issue.

**Comment by heartsentwined** [comment]
`emberModel` changeset [here](https://github.com/heartsentwined/ember-auth/commit/e71433ffa97e5287efceb671cc195c5e78d96837#lib/module/ember-model.em).

**Comment by flynfish** [comment]
After further testing, 7.1.1 works correctly too, so the issue must be with `$.url` / `purl` library update. I am not using `urlAuthenticatable` module however. Any ideas?

**Comment by heartsentwined** [comment]
Oh, thanks and sorry for that, it actually unearthed a stupid bug - the `v7.1.1` release is bogus. The source files do contain the changeset, but the actual dist files haven't been updated from `v7.1.0`. I'll yank the `v7.1.1` release.

**Comment by flynfish** [comment]
Ah ok. I can confirm that #84 is the culprit for the break then. I will try and figure out why..

**Comment by flynfish** [comment]
@heartsentwined is the issue something to do with the `settings.data` being `JSON.stringify`'d and ember-auth therefore can't add the `auth_token`? 

here is the output I see when viewing the settings object in the `_ajax` function while debugging:

7.1.0 version output:

```
Object {url: "http://localhost:4000/subscriptions/singleton.json", type: "PUT",dataType: "json", data: Object}
```

7.1.2 version output:

```
Object {url: "http://localhost:4000/subscriptions/singleton.json", type: "PUT", dataType: "json",contentType: "application/json; charset=utf-8", data: "{"subscription":{"id":"singleton","card_type":"Dis…","status":"active","name":"Kyle","plan":6}}"}
```

**Comment by heartsentwined** [comment]
Good catch, that might be the case, I'll ask over at the original PR author, and see if a `stringifiy`'d string is crucial to `ember-model`'s API.

**Comment by heartsentwined** [comment]
@flynfish could you do a test on whether removing the `stringify` would indeed fix the problem?

**Comment by flynfish** [comment]
I get an error removing the stringify. I tried it yesterday, but I'm not by a computer at the moment to give more details

**Comment by heartsentwined** [comment]
Pending further update from @flynfish. The PR to remove `stringify` is already there (passing). :smile: (#94)

**Comment by heartsentwined** [comment]
Fix at #95 merged into master and `7.1.3`.
